backend/sync-mounter-service/src/jvmMain/kotlin/services/MountService.kt

```diff
@@ -0,0 +1,128 @@
+package dk.sdu.cloud.sync.mounter.services
+
+import dk.sdu.cloud.calls.RPCException
+import dk.sdu.cloud.sync.mounter.SyncMounterConfiguration
+import dk.sdu.cloud.sync.mounter.api.*
+import io.ktor.http.*
+import java.io.File
+import java.nio.file.Paths
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlin.io.path.ExperimentalPathApi
+import kotlin.io.path.pathString
+import kotlin.io.path.readSymbolicLink
+
+class MountService(
+    val config: SyncMounterConfiguration,
+    val ready: AtomicBoolean
+) {
+    private val MS_BIND: Long = 4096
+    private val O_NOFOLLOW = 0x20000
+
+    @OptIn(ExperimentalPathApi::class)
+    fun mount(request: MountRequest) {
```

Comment:
Same issue with `startsWith`. You are not checking if they are a descendant but just checking if the strings starts with another. You end up with this issue:

```
>>> java.io.File("/mnt/cephfssomethingelse").canonicalPath.startsWith("/mnt/cephfs")
res2: kotlin.Boolean = true
```

Reply:
The joinPath already adds a ending slash to the path. I've updated the paths from the configuration to always end with a slash.